### ISSUE TITLE
The prop value with an expression type of OptionalMemberExpression could not be resolved

### ISSUE THREAD

**Issue by HsuTing** [jsx-eslint/jsx-ast-utils#70]
I use `@babel/plugin-proposal-optional-chaining` like this:
```js
<div href={data?.href} />
```
I found only `href` will throw error. Other props will not throw error, like this:
```js
<div test={data?.test} />
```

**Comment by pdong** [comment]
I am seeing this with `.title` as well 

`{data?.title}` will error out with

```
The prop value with an expression type of OptionalMemberExpression could not be resolved.
  Please file issue to get this fixed immediately.
Error: The prop value with an expression type of OptionalMemberExpression could not be resolved.
  Please file issue to get this fixed immediately.
    at Object.extractLiteral [as JSXExpressionContainer] (/Users/pdong/node_modules/jsx-ast-utils/lib/values/expressions/index.js:202:11)
    at getLiteralValue (/Users/pdong/node_modules/jsx-ast-utils/lib/values/index.js:61:35)
    at extractValue (/Users/pdong/node_modules/jsx-ast-utils/lib/getPropValue.js:24:12)
    at getLiteralPropValue (/Users/pdong/node_modules/jsx-ast-utils/lib/getPropValue.js:56:10)
    at JSXAttribute (/Users/pdong/node_modules/eslint-plugin-jsx-a11y/lib/rules/aria-proptypes.js:88:58)
    at listeners.(anonymous function).forEach.listener (/Users/pdong/node_modules/eslint/lib/util/safe-emitter.js:47:58)
    at Array.forEach (<anonymous>)
    at Object.emit (/Users/pdong/node_modules/eslint/lib/util/safe-emitter.js:47:38)
    at NodeEventGenerator.applySelector (/Users/pdong/node_modules/eslint/lib/util/node-event-generator.js:251:26)
    at NodeEventGenerator.applySelectors (/Users/pdong/node_modules/eslint/lib/util/node-event-generator.js:280:22)
```

but if I change it it works. 🤔

**Comment by jessebeach** [comment]
Working on it now. I need to update all the dev dependencies to get there :)

**Comment by jessebeach** [comment]
@HsuTing , Fixed in version v2.2.1. This will log an error in the console, but it will no longer throw.

**Comment by HsuTing** [comment]
@jessebeach OK, thank you for your help.

**Comment by MathiasKandelborg** [comment]
Could we not get this error? The message comes from here, following the issue trail from eslint.

If there is an error/bug, it should get fixed, if it's a false report, I don't want to see it.

**Comment by jessebeach** [comment]
@MathiasKandelborg I understand the frustration here. I feel it, too. I hate seeing these bugs pop up. I'll look into it this weekend and put out a patch release.

**Comment by ljharb** [comment]
note that v3.0 of jsx-ast-utils is out, and eslint-plugin-jsx-a11y has been updated with it, but not yet released - so a patch release here would still need a release in eslint-plugin-jsx-a11y.

**Comment by ljharb** [comment]
Pretty sure this is fixed by #77. eslint-pliugin-react has already been updated with it, and eslint-plugin-jsx-a11y will get it on the next release.
